### PR TITLE
Added Client Tests and Updated Workflow to not cache the Test files

### PR DESCRIPTION
**Pull request by Unic-X**
@bsach64 Review !

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -23,39 +23,42 @@ func ConnectToServer(user, password, host string) (*ssh.Client, error) {
 	return client, nil
 }
 
-func Upload(client *ssh.Client, f string) {
+func Upload(client *ssh.Client, f string) error {
 	sftpClient, err := sftp.NewClient(client)
 	if err != nil {
 		log.Fatalf("Failed to create SFTP client: %v", err)
+		return err
```
Comment (bsach64) [inline client/client.go:30]: log.Fatalf exits the program completely. Cannot return after call to fatalf https://pkg.go.dev/log#Fatalf

baaki jagah use kiya hai usko bhi change karde

File: client/client_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+func testConnection(t *testing.T) {
+
+    _, err := ConnectToServer(
+      "test_user",
+      "password",
+      "127.0.0.1:2022",
+    )
+
+     if err!=nil{
```
Comment (bsach64) [inline client/client_test.go:28]: not formatted correctly 

gofmt -s -w .

(Iska matlab apna CI broken hai)

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -18,6 +18,9 @@ jobs:
       with:
         go-version: '1.22.5'
 
+    - name : Clean
```
Comment (bsach64) [inline .github/workflows/go.yml:21]: Iski kya need hai?

**Comment by bsach64** [comment]
private/id_rsa kya karta hai?

**Comment by Unic-X** [comment]
> private/id_rsa kya karta hai?
Clients do not connect to servers without private key
image

**Comment by Unic-X** [comment]
CI galat hai kyuki without pvt key it should fail

**Comment by bsach64** [comment]
> CI galat hai kyuki without pvt key it should fail 

Me toh formatting ki baat kar raha tha joki #8 se fix hogi.

CI kabhi binary ko run nhi karta hai. Bas tests run hote hai

**Comment by Unic-X** [comment]
> > CI galat hai kyuki without pvt key it should fail
> 
> Me toh formatting ki baat kar raha tha joki #8 se fix hogi.
> 
> CI kabhi binary ko run nhi karta hai. Bas tests run hote hai

To server ke liye bhi test add krdu?

**Comment by bsach64** [comment]
> > > CI galat hai kyuki without pvt key it should fail
> 
> > 
> 
> > Me toh formatting ki baat kar raha tha joki #8 se fix hogi.
> 
> > 
> 
> > CI kabhi binary ko run nhi karta hai. Bas tests run hote hai
> 
> 
> 
> To server ke liye bhi test add krdu?
> 
> 
> 
> Aur build.sh likh deta hu format aur build aur test sab kr dega local check ke liye

Kar sakta hai

**Comment by Unic-X** [comment]
Ab sahi hai fail ho raha bina key ke

**Comment by Unic-X** [comment]
Client ke liye bhi struct banana tha fir same structure follow krte hue karu to 💀

File: client/client.go [hunk 4]
```diff
@@ -27,36 +27,31 @@ func Upload(client *ssh.Client, f string) error {
 	sftpClient, err := sftp.NewClient(client)
 	if err != nil {
 		log.Fatalf("Failed to create SFTP client: %v", err)
-		return err
```
Comment (bsach64) [inline client/client.go]: err is always nil
consider:
```
log.Printf(...)
return err
```

**Review by bsach64 — APPROVED** [review]

**Comment by bsach64** [comment]
> Client ke liye bhi struct banana tha fir same structure follow krte hue karu to 💀

Abhi ke liye thik hai

**Comment by Unic-X** [comment]
Krra merge fir

**Pull request merged**